pelican/log.py

```diff
@@ -87,15 +87,24 @@ class FatalLogger(LimitLogger):
     warnings_fatal = False
     errors_fatal = False
 
-    def warning(self, *args, **kwargs):
-        super().warning(*args, **kwargs)
+    def callHandlers(self, record):
+        """
+        Raise RuntimeError if log level is higher than acceptable.
+
+        Level limit set via the ``--fatal`` flag when running pelican.
+        """
+        super().callHandlers(record)
+        level_limit = None
```

Comment:
For such a small scope, maybe it's redundant.  However, I'm wary of initializing variables inside branching logic when that name is referenced again outside that block, as it can lead to bugs when code gets modified...

I'll take another look at this and see if there's a more elegant way to handle it, thanks!

Reply:
Generally true, but once you chance the comparison to `>=`, defining a default as `None` would not help since it would raise an error when compared to an `int` anyway :).